returnn/util/file_cache.py

```diff
@@ -312,6 +326,26 @@ def _check_existing_copied_file_maybe_cleanup(src_filename: str, dst_filename: s
         return True
 
 
+def _copy_with_prealloc(src: str, dst: str):
+    dst_size = os.stat(src).st_size
+    with open(dst, "wb") as dst_file:
+        if dst_size > 0:
+            # We prealloc size + 1 to avoid having a file that has the exact same
+            # size as its original. After copying, we then truncate the additional
+            # byte away. This allows detecting incompletely copied files.
+            #
+            # See also `_check_existing_copied_file_maybe_cleanup`.
```

Comment:
```suggestion
            # We prealloc size + 1 to avoid having a file that has the exact same size
            # as its original, to allow detecting incompletely copied files,
            # as done in `_check_existing_copied_file_maybe_cleanup`.
            # After copying, we then truncate the additional byte away.
```